### ISSUE TITLE
v4.0.0-beta.4: Regression from v4.0.0-beta.3: `ERR_UNSUPPORTED_DIR_IMPORT` Directory import '\node_modules\primevue\api' is not supported resolving ES modules when using vitest

### ISSUE THREAD

**Issue by sceee** [primefaces/primevue#5821]
### Describe the bug

After upgrading from `v4.0.0-beta.3` to `v4.0.0-beta.4`, the following error occurs when executing tests using vitest that worked previously in `v4.0.0-beta.3`:
```
Directory import '\node_modules\primevue\api' is not supported resolving ES modules imported from \node_modules\primevue\config\config.mjs Did you mean to import primevue/api/api.mjs?
```

It seems the issue results from this import of `primevue/api` here as this is a directory import:
https://github.com/primefaces/primevue/blob/401dd225677796f05321b0c98ac68b51f3487de6/components/lib/config/PrimeVue.js#L1

In the build, this directory `\node_modules\primevue\api` contains a `package.json` with the following content (changed in 1d3096d74e63876514d0485af968717a99c94836 and a438dbae1eaecc42e6b692e9adeb0a658c4afff4 ):
```json
{
    "main": "./api.mjs",
    "module": "./api.mjs",
    "types": "./Api.d.ts"
}
```
This seems to cause issues as the ESM loader used by vitest does not support this.

I was _guessing_ a potential fix could be to properly declare a proper [export map](https://nodejs.org/api/packages.html#conditional-exports) instead here so ESM loaders can pick up the correct file. Something like:
```json
{
    "exports": {
        ".": {
          "types": "./Api.d.ts",
          "import": "./api.mjs",
          "require": "./api.cjs.js"
        }
    }
}
```
(Side note: In this directory there is also a `api.esm.js` which seems to be identical to the `api.mjs` except for the sourcemap comment.)

However, I could not get it running by simply changing the `package.json` in the `\node_modules\primevue\api` directory. 

Unfortunately, this issue makes primevue v4.0.0-beta.4 impossible to be tested using vitest, therefore blocks the update in my case.

### Reproducer

https://stackblitz.com/edit/primevue-create-vue-typescript-issue-template-ctejwg?file=package.json

### PrimeVue version

4.0.0-beta.4

### Vue version

3.x

### Language

TypeScript

### Build / Runtime

Vite

### Browser(s)

node / vitest

### Steps to reproduce the behavior

1. Open the attached stackblitz
2. Potentially stop the execution using Ctrl+C in the terminal
3. Run npm test in the terminal to "execute" the test

This shows the error.

### Expected behavior

Primevue components can be tested using vitest as it used to work until primevue `v4.0.0-beta.3`.
No error during tests.

**Comment by sceee** [comment]
Ok, I found a workaround by adding the following to the `vite.config.ts`:
```typescript
test: {
    ...,
    server: {
      deps: {
        inline: ['primevue'],
      },
    },
  },
```

According to the `vitest` docs, [this could be helpful to handle packages that ship `.js` in ESM format (that Node can't handle)](https://vitest.dev/config/#server-deps-inline).

However, I don't think it is intended that the code is shipped in such a way and therefore such an option is required to use primevue in tests especially since this was working in `v4.0.0-beta.3` without that option.

So I would appreciate a proper solution in primevue 😃 .

**Comment by cmcnicholas** [comment]
I get a similar issue importing primevue/toast (working beta3, failing beta4 in vitest)

`Error: Directory import 'xxx\node_modules\primevue\portal' is not supported resolving ES modules imported from xxx\node_modules\primevue\toast\toast.mjs
Did you mean to import primevue/portal/portal.mjs?`

**Comment by sceee** [comment]
This seems to be resolved by the latest bundle changes in [4.0.0-rc.1](https://github.com/primefaces/primevue/releases/tag/4.0.0-rc.1).
